api/v1beta1/octavia_types.go

```diff
@@ -201,6 +204,10 @@ type OctaviaSpecBase struct {
 	// Apache Container Image URL
 	ApacheContainerImage string `json:"apacheContainerImage"`
 
+	// +kubebuilder:validation:Optional
+	// Octavia API timeout
+	APITimeout string `json:"apiTimeout,omitempty"`
```

Comment:
it should have been an int

Reply:
Let's do a quick follow-up to adjust?

Reply:
yeah, I implemented as int from the beginning but after comparing how neutron-operator (directly managing as string) and how other did (cinder, manilla, glance managing as int) I saw that the annotation is mapping as string, doing a conversion from the ones using as base a int, so maybe I decide the wrong way, so feel free to comment here and I will change to int in a follow-up PR. wdyt?

Reply:
ok then it makes sense to have a string there

Reply:
why do we need an api parameter for this? the rout is generated by the openstack-operator. in the service-operator we should just default to e.g. the 120s. via the openstack-operator route overrides you can customize it to whatever value it should be used different then 120s. This is how it was done for neuron

https://github.com/openstack-k8s-operators/octavia-operator/pull/376
https://github.com/openstack-k8s-operators/openstack-operator/pull/722

Reply:
What I mean is there is a generic interface on how to customize the timeout of a route. we should not introduce specific parameters for which there is a way to do customization. the service operator should just set the default to whatever is appropriate. Therefore I think we should do a partial revert to remove the APITimeout parameter unless there is a reason to have it

Reply:
ok, I see that it we have it in e.g. glance with https://github.com/openstack-k8s-operators/glance-operator/pull/550 , but to make sure the timeout also gets set on the httpd config. so this probably should have a follow up to get it configured there. and we need to get it applied across all operators. in general we just need to be careful to not introduce parameters for settings which can be applied via an interface which is already available.

Reply:
> why do we need an api parameter for this? the rout is generated by the openstack-operator. in the service-operator we should just default to e.g. the 120s. via the openstack-operator route overrides you can customize it to whatever value it should be used different then 120s. This is how it was done for neuron
> 
> [openstack-k8s-operators/neutron-operator#324](https://github.com/openstack-k8s-operators/neutron-operator/pull/324) [openstack-k8s-operators/openstack-operator#722](https://github.com/openstack-k8s-operators/openstack-operator/pull/722)

yeah, I did taking a look to exactly those two PR to try to do exactly the same in the octavia-operator